### PR TITLE
Make "standard" e2e scenario target work

### PR DESCRIPTION
**Pull request by 0sewa0**
## Description

Currently when running the `standard` e2e test scenario: (due to bug in the e2e framework)
```
=== Skipped
=== SKIP: test/scenarios/standard TestStandard (1.22s)
    default.go:58: istio enabled: false
    env.go:211: Skipping feature "dynakube in network zone": matched label provided in --skip-lables "name=[cloudnative-network-zone]"
```

Changes: (for details, look at the commits)
- [Make "standard" e2e scenario target work](https://github.com/Dynatrace/dynatrace-operator/commit/1133765e97c91e993c1d02eaee64418a7942e9da)
- [Make e2e test cases individually callable](https://github.com/Dynatrace/dynatrace-operator/commit/89392263b213a870a27d05b4bffa3861d198ac34)
- Fix broken tests in the `standard` scenario, duo to not being run for several months
   -  [Fix label-version e2e test](https://github.com/Dynatrace/dynatrace-operator/commit/8ac8c87bbf59117b5be7c440500eed8b2a060ca3) ([2. attempt](https://github.com/Dynatrace/dynatrace-operator/pull/3719/commits/cb04247703d3e8b75b62e2697efe5a0fa0f64b87))
   - [Fix codemodules-image e2e test](https://github.com/Dynatrace/dynatrace-operator/commit/3e1fdc9944db8f2e753abfc4d51f55dde1eb30a0)
   - [Fix init-container check](https://github.com/Dynatrace/dynatrace-operator/pull/3719/commits/5edaa613aa447378955e712e185eabf9ee916925)
   - [Fix withoutcsi e2e test](https://github.com/Dynatrace/dynatrace-operator/pull/3719/commits/7e3d9a84e5c7e8b0bd934553476f2ee0461c86a9)
   - [Fix "disabledautoinjection" e2e test](https://github.com/Dynatrace/dynatrace-operator/pull/3719/commits/78feb9ac6f94b5ae0fccd90b6a2a82448811e680)
- [Stop DDOSing the public ecr to get latest versions](https://github.com/Dynatrace/dynatrace-operator/pull/3719/commits/fc045f14349659ccb866d90f866930737edd3022)

## How can this be tested?

Run:
```
make test/e2e/standard
```

Run all (or as many as you can) of the individual e2e targets, example:
```
make test/e2e/cloudnative/resilience
```

### CODE REVIEW

File: test/scenarios/istio/istio_test.go [hunk 1]
```diff
@@ -52,5 +56,18 @@ func TestIstio(t *testing.T) {
 		codemodules.WithProxyAndAGCert(t, proxy.ProxySpec),
 		codemodules.WithProxyCAAndAGCert(t, proxy.HttpsProxySpec),
 	}
-	testEnv.Test(t, feats...)
+
+	filteredFeats := []features.Feature{}
+
+	if cfg.FeatureRegex() != nil {
```
Comment (waodim) [inline test/scenarios/istio/istio_test.go:62]: Can we extract this part as it is used several times and seems like it is the same (if I am not mistaken)?

Comment (0sewa0) [reply test/scenarios/istio/istio_test.go:62]: on it

**Review by andriisoldatenko — APPROVED** [review]
code LGTM, still running some e2e tests in my cluster. Just in case of you want to merge it today before weekends :)

File: hack/make/tests/e2e.mk [hunk 2]
```diff
@@ -26,7 +26,7 @@ test/e2e-publish:
```
Comment (waodim) [inline hack/make/tests/e2e.mk:26]: Although code LGTM, after running the standard scenario several times I still got some errors. I am not sure whether this is only me and my local set up, so please if anyone can confirm that they are also getting errors, reply here.

Example (this time it is the withoutcsi test):
```
    app.go:151: 
                Error Trace:    /Users/albian.krasniqi/Library/CloudStorage/OneDrive-Dynatrace/Desktop/Proj/dynatrace-operator/test/helpers/sample/app.go:151
                                                        /Users/albian.krasniqi/go/pkg/mod/sigs.k8s.io/e2e-framework@v0.4.0/pkg/env/env.go:444
                                                        /Users/albian.krasniqi/go/pkg/mod/sigs.k8s.io/e2e-framework@v0.4.0/pkg/env/env.go:483
                Error:          Received unexpected error:
                                client rate limiter Wait returned an error: context deadline exceeded
                Test:           TestStandard/app-monitoring-without-csi/install_already_injected_sample_app
=== NAME  TestStandard/app-monitoring-without-csi
    app.go:178: 
                Error Trace:    /Users/albian.krasniqi/Library/CloudStorage/OneDrive-Dynatrace/Desktop/Proj/dynatrace-operator/test/helpers/sample/app.go:178
                                                        /Users/albian.krasniqi/go/pkg/mod/sigs.k8s.io/e2e-framework@v0.4.0/pkg/env/env.go:444
                                                        /Users/albian.krasniqi/go/pkg/mod/sigs.k8s.io/e2e-framework@v0.4.0/pkg/env/env.go:507
                Error:          Received unexpected error:
                                deployments.apps "random-user" not found
                Test:           TestStandard/app-monitoring-without-csi
--- FAIL: TestStandard (424.41s)
```

**Pull request merged**